core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java

```diff
@@ -223,4 +229,35 @@ public void testInvalidIndexName() {
                     e.getMessage().contains("Invalid index name [..], must not be \'.\' or '..'"), equalTo(true));
         }
     }
+
+    /**
+     * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
```

Comment:
I suppose it doesn't have to be, but I wanted to test not just the setting changing but that the replication operation uses the new setting... so this was the most convenient way to achieve that.  I have a similar test for index creation in `CreateIndexIT`